### DISCUSSION TITLE
Exercise 2 Part 2 - Expected Training Times

### DISCUSSION THREAD

**Discussion by BananaLee** [sebastian-hofstaetter/teaching#104] [Advanced IR: Exercise Q&A] [answered]
Hi there,
Just so I know if I'm on the right track or not - I'm running Colab Pro and have passed everything to the cuda device for this. But a single epoch is taking slightly less than an hour to train. That means, in order to properly validate and test it, I would need quite a substantial amount of time (more then 24 hours for 25 epochs). 

Before I push the button on making a long run, can I confirm if I'm on the right track time-wise or if I have missed something? Is there a rough indication of how many epochs one should expect to reach an optimal solution and how long each epoch should take?

Cheers,
Ben

**Comment by sophiaalthammer** [comment, marked as answer]
Hi Ben,

1 hour for one epoch sounds about right, if you included early stopping based on the validation set and you are using the recommended hyperparameters, the training should resume after 1/2 epochs.

Best,
Sophia
